features/FEATURE_BLE/source/generic/GenericGap.tpp

```diff
@@ -2509,6 +1786,8 @@ ble_error_t GenericGap<PalGapImpl, PalSecurityManager, ConnectionEventMonitorEve
         return BLE_ERROR_INVALID_PARAM;
     }
 
+    prepare_legacy_advertising_set();
```

Comment:
Is it necessary to call that function if the advertising handle is not the legacy one ?

Reply:
I'll add a check for legacy handle